core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -655,15 +656,13 @@ public SuggestStats suggestStats() {
 
     public CompletionStats completionStats(String... fields) {
         CompletionStats completionStats = new CompletionStats();
-        final Engine.Searcher currentSearcher = acquireSearcher("completion_stats");
-        try {
+        try (final Engine.Searcher currentSearcher = acquireSearcher("completion_stats")) {
             PostingsFormat postingsFormat = PostingsFormat.forName(Completion090PostingsFormat.CODEC_NAME);
             if (postingsFormat instanceof Completion090PostingsFormat) {
                 Completion090PostingsFormat completionPostingsFormat = (Completion090PostingsFormat) postingsFormat;
                 completionStats.add(completionPostingsFormat.completionStats(currentSearcher.reader(), fields));
             }
-        } finally {
-            currentSearcher.close();
+            completionStats.add(CompletionFieldStats.completionStats(currentSearcher.reader(), fields));
```

Comment:
Not sure I understand, before we had the finally clause but with the change I removed the finally clause and used try-with-resources on the engine.searcher?